client/state/notices/types.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+export type NoticeId = string;
+
+export type NoticeStatusError = 'is-error';
+export type NoticeStatusInfo = 'is-info';
+export type NoticeStatusPlain = 'is-plain';
+export type NoticeStatusSuccess = 'is-success';
+export type NoticeStatusWarning = 'is-warning';
+export type NoticeStatus =
+	| NoticeStatusError
+	| NoticeStatusInfo
+	| NoticeStatusPlain
+	| NoticeStatusSuccess
+	| NoticeStatusWarning;
+
+export type NoticeText = string | ReactNode;
+
+// Notice options that are always the same
+export interface BaseNoticeOptions {
+	button?: string;
+	displayOnNextPage?: boolean;
+	duration?: null | number;
+	href?: string;
+	isPersistent?: boolean;
+	onClick?: () => void;
+	showDismiss?: boolean;
+}
+
+// Notice options as they're expected by the action creator
+export interface NoticeOptions extends BaseNoticeOptions {
+	id?: NoticeId;
+}
+
+// Notice options as they're returned by the action creator
+export interface NoticeActionOptions extends BaseNoticeOptions {
+	noticeId: NoticeId;
+	status: NoticeStatus;
+	text: NoticeText;
+}
+
+// Notice action as returned by the action creator
+export type NoticeAction = {
+	type: string;
+	notice: NoticeActionOptions;
+};
```

Comment:
Can this also be generic on the `type` property?

Reply:
See https://github.com/Automattic/wp-calypso/pull/49461#discussion_r567926811 for my rationale on that.